### DISCUSSION TITLE
The :then Output Is Wrong or Compare Works Wrongly

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#8226] [General]
* open https://tiddlywiki.com
* create a new tiddler with below script

```xml
<$list
 filter="[all[current]!is[system]] :then[all[current]links[]count[]compare:integer:gteq[1]]">

I am here! -- <<currentTiddler>>
</$list>
```

It shows `I am here! – New Tiddler 1` while it should display nothing!

If you change the code as below (note to the value in `gteq[0]`)

```xml
<$list
 filter="[all[current]!is[system]] :then[all[current]links[]count[]compare:integer:gteq[0]]">

I am here! -- <<currentTiddler>>
</$list>
```

It shows `I am here! – 0` This is correct!

**Comment by kookma** [comment]
I read the doc again and it seems the above code works correctly. The doc says:

If the output of a :then prefixed filter run is itself an empty list, the result of all previous filter runs is passed through unaltered.

So my code above works correctly! Sorry for confusion.

**Reply by Jermolene** [reply]
No worries @kookma – could the documentation be improved to reduce the confusion?

**Reply by kookma** [reply]
Yes, I think the point: "If the output of a `:then` prefixed filter run is itself an empty list, the result of all previous filter runs is passed through unaltered." should be emphasized and one example can be added. I can submit a PR.
